docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,230 @@
+repositories, we have a [Nerves Radar] project in our GitHub Organization.
+Unfortunately, this isn't visible to the public because GitHub requires that you
+be a member of the Organization to see Organization-level Projects. If you want
+to get involved in this process, get in touch with us on Slack!
+
+We currently don't have this process formalized, so feel free to contribute to
+this document itself!
+
+[Nerves Radar]: https://github.com/orgs/nerves-project/projects/1
+
+## Code and Documentation changes
+
+If you're looking for a way to directly contribute to the code and
+documentation, check our [repositories] for Issues that are labeled with
+`kind:documentation`, `kind:bug`, or `kind:enhancement`. You can use the
+`level:stater`, `level:intermediate`, and `level:advanced` labels to determine
+how difficult we think it will be, whether you're looking for an easy first-time
+contribution or an extra challenge.
+
+If you're working on an Issue, feel free to set yourself as the **Assignee**,
+and don't feel bad about removing yourself if you later decide that you're not
+going to work on it. This helps us keep track of which Issues are currently
+being worked-on, avoiding duplicated effort and Issues that get stale over time.
+
+Working on your first Pull Request? You can learn how from this *free* series,
+[How to Contribute to an Open Source Project on GitHub].
+
+The core team has GitHub integrations with Slack, so we will be notified
+immediately in the #nerves-dev channel about activity on GitHub Issues and Pull
+Requests. We really appreciate you taking the time to record any Issues in
+GitHub so that we don't lose track, but we all have busy lives outside of Nerves
+work, so don't take it personally if it takes us a long time to respond.
+
+[How to Contribute to an Open Source Project on GitHub]: https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github
+
+### Licensing and copyright assignment
+
+Copyright and licensing are very important to open-source software projects.
+Nerves components fall under several licenses, so please review the license for
+the project to which you are contributing. By contributing, you certify that you
+have the right to submit the changes under the project's open source license and
+assign copyright to the Nerves Project maintainers.
+
+### Style guide for documentation
+
+This is currently a work-in-progress. Feel free to suggest any missing details!
+
+#### Step-by-step command-line instructions
+
+We want them to be very user-friendly, even for people who are not familiar with
+using a command-line interface. Therefore, we will not show the `$` or `#`
+prompt before the commands to be entered to make them easier to copy and paste
+into a terminal. For example:
+
+```bash
+export MIX_TARGET=rpi3
+mix firmware
+mix firmware.burn
+```
+
+When showing what we expect a person to see or do on the command-line, we will
```

Comment:
If I understand correctly, the logic is "if output is shown, then show prompt, else don't show prompt". Therefore, if we showed one line of output in the example above, we'd require a `$` in front of all of the commands. Does this seem weird or is there more to the logic?

Reply:
I'm not sure which "example above" you're talking about since the three lines above are all input without any output shown. My feeling is that we'd include all of the prompts or none of them in a given block of pre-formatted text.

I agree that by the logic we currently have here, you'd include all the prompts if there's only one line of output. If you thought that looked weird, you could potentially just not include the line of output if it's not necessary for understanding. I think in general it's best not to pick and choose which commands are showing output vs. only showing the commands to be run within a given block of text because someone might think there's a problem if there's a bunch of text generated for them as opposed to nothing as shown in the example. You could also split the text into multiple blocks, like "run these three commands", then "run this command, which will have output that looks like this," then "run a few more commands", etc.

If you see a way that we can make things more clear or want to change the rules, I'm totally fine with that. I just wanted to get whatever we have nailed-down so that we can build on it.